routes/api.js

```diff
@@ -1,29 +1,117 @@
 var express = require('express');
 var db = require('../database/controllers/dbm');
+var config = require('../local_modules/config')
+var cache = require('../lib/cache');
 var router = express.Router();
 
-// log request
-router.use(function(req, res, next) {
-    console.log('%s %s', req.method, req.url);
-    next(); 
-});
+var timer = config.get('General', 'badge_override_interval')
 
 router.post('/user-access', function(req, res) {
-    var stationState = 'Enabled';
-    if(!req.body || req.body.length === 0 || req.body.length > 200) {
-        res.set({
-            'Content-Typer': 'text/plain',
-            'Station-State': stationState = 'Disabled'
-        });
-        res.status(403).send('Forbidden');
-    } else {
-        res.set({
-            'Content-Type': 'text/plain',
-            'User-Id-String': req.body.id,
-            'Station-State': stationState
-        });
-        res.status(200).send('OK');
-    }        
+	if(!req.body || req.body.length === 0 || req.body.length > 200) {
+		// body sanity check
+		res.set({
+				'Content-Type': 'text/plain',
+				'Station-State': 'Disabled'
+		});
+		res.sendStatus(403);
+	};
+
+	if(!req.headers.badge || !req.headers.sid || !req.headers['station-state']) {
```

Comment:
There's no `sid` header in the API spec. The header name you want is `station-id`. This appears in numerous places in the code, they should all be fixed. Also needs to be fixed in `tests/api_test.js`